doc/mclag/MCLAG_Enhancements_HLD.md

```diff
@@ -0,0 +1,810 @@
+	- [9.2 FDB](#9_2-FDB)
+	- [9.3 Config handling](#9_3-Config-handling)
+	- [9.4 Port Isolation](#9_4-Port-Isolation)
+	- [9.5 MCLAG Interface Flap](#9_5-MCLAG-Interface-Flap)
+	- [9.6 MCLAG Unique IP enhancements](#9_6-MCLAG-Unique-IP-enhancements)
+- **[10 Internal Design Information](#10-Internal-Design-Information)**
+
+# List of Tables
+
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev  | Date       | Author                                   | Change Description |
+| ---- | ---------- | ---------------------------------------- | ------------------ |
+| 0.1  | 08/02/2019 | Lisa Nguyen, Praveen Elagala, Abhishek Dharwadkar, Sabarivel Sakthivel, Adam Yeung | Initial draft      |
+| 0.2  | 10/22/2019 | Tapash Das                               | L3 Protocol support enhancement |
+
+# About this Manual
+This document provides general information about SONiC MCLAG feature enhancements. The original MCLAG  feature is described at https://github.com/Azure/SONiC/blob/f478fe7cbc03c144f3b147e9638f460f764ce4b7/doc/Sonic-mclag-hld.md 
+
+
+# Definition/Abbreviation
+## Table 1: Abbreviations
+| **Term** | **Definition**                       |
+| -------- | ------------------------------------ |
+| MCLAG    | Multi-Chassis Link Aggregation Group |
+| ICCP     | Inter-Chassis Communication Protocol |
+| FDB      | Layer-2 (MAC) based forwarding table |
+| MHD      | Multi Home Device                    |
+
+# 1 Feature Overview
+
+This document captures the feature enhancements of SONiC ICCP MCLAG. This includes MCLAG configuration support, data structure changes, MAC event handling optimizations for scaling performance, support of static MAC address over MCLAG, support bridge-port isolation group for BUM control to MCLAG, and traffic recovery sequencing for traffic loop prevention. 
+
+## 1.1 Functional Requirements
+
+### 1.1.1 Static MAC support over MCLAG
+- Advertise locally configured Static MAC address to peer MCLAG node
+- Withdraw MAC from peer MCLAG node when static MAC address is un-configured
+- Dynamic MAC move is prohibited on Static MAC over MCLAG
+
+### 1.1.2 Configuration and Management Requirements
+- Configure MCLAG domain and corresponding attributes domain-id, local-ip-address and peer-ip-address
+
+- Configure optional attribute [peer-interface] 
+
+- Configure keep-alive timer value
+
+- Configure session-timeout value
+
+- Add support for addition/deletion of mclag_interfaces under MCLAG domain
+
+### 1.1.3 Performance and Scalability Requirements
+- Optimize code flow and data structures to improve scaling performance
+- Support 40K FDB entries
+- Support 4K VLAN's
+
+### 1.1.4 Warm Boot Requirements
+- MCLAG peer nodes should reconcile the local FDB table upon completion of warm boot, as the MAC learn and age updates from peer would be lost during the time ICCP control session is down due to warm boot.
+
+### 1.1.5 Unique IP for supporting L3 protocol over MCLAG VLAN interface Requirements
```

Comment:
what is this unique IP requirement? is there an example?

Reply:
Lets assume MCLAG Interface is member of Vlan 5.  In this case, IP and MAC on Vlan interface in both MCLAG nodes are same. 

MCLAG Node 1 (Active)    : Vlan5 : IP_1 -- MAC_A
MCLAG Node 2 (Standby) : Vlan5 : IP_1 -- MAC_A
MCLAG Client : Vlan5 : IP_3 -- MAC_C

With this we cannot associate L3 Protocols (BGP, BFD etc) with IP_1.

Unique IP requirement will allow L3 Protocols on Vlan5.

Reply:
3.6.1.1 Local MAC upstream using STATE_DB FDB_TABLE
Please refer to PR 1259：MCLAG sync FDB MAC from STATE_DB #1259[https://github.com/Azure/sonic-swss/pull/1259](url)

Reply:
@jianjundong  Thanks for pointing to the PR 1259 . Will delete the relevant sections from Enhancements HLD.
Can you please share when can we expect the change to be merged.?

Reply:
In PR 1259, I commented it must work with PR 'Layer 2 Forwarding Enhancements #885'. In currently implementation, STATIC MAC configured by CLI and MAC MOVE are not supported, they are enhanced in PR 885. To dynamic MAC learnt by ASIC, the changes of PR 1259 are test OK. In theory, mclagsyncd is the consumer of STATE_DB FDB_TABLE, does not care the MAC type.
Merging can be performed automatically with 1 approving review.

Reply:
@jianjundong Thanks for the information shared. Understand the dependency. We will also perform the review on the PR#1259 and provide feedback if any in next couple of days.